system-tests/resources/cli-tools/entrypoint.sh

```diff
@@ -3,34 +3,92 @@
+          sleep 1
+      fi
+      retryCount=$((retryCount+1))
+  done
+  if [ "$onboardingCompleted" == false ]; then
+      echo "$participantDid onboarding is not completed yet. Exiting..."
+      exit 1
+  fi
+}
+
+function registerParticipant() {
+  local participantName="$1"
+  local participantDid="$2"
+
+  echo "Registering $participantName"
+  java -jar registration-service-cli.jar \
+              -d="did:web:did-server:registration-service" \
+              --http-scheme \
+              -k=/resources/vault/$participantName/private-key.pem \
+              -c="$participantDid" \
+               participants add
+}
+
+function seedVerifiedCredentials() {
+  local participantName="$1"
+  local participantDid="$2"
+  local region="$3"
+
+   for subject in '"region": "'$region'"' '"gaiaXMember": "true"'
+   do
+     echo "Seeding VC for $participantName: $subject"
+     vcId=$(uuidgen)
+     java -jar identity-hub-cli.jar \
+                 -s="http://$participantName:8181/api/identity-hub" \
+                 vc add \
+                 -c='{"id":"'$vcId'","credentialSubject":{'"$subject"'}}' \
+                -b="$participantDid" \
+                 -i="did:web:did-server:gaia-x" \
+                 -k="/resources/vault/gaia-x/private-key.pem"
+   done
+}
+
 PARTICIPANTS=(company1:eu company2:eu company3:us)
 
-# Register dataspace participants
 for participant in "${PARTICIPANTS[@]}"; do
     participantArray=(${participant//:/ })
 
     participantName=${participantArray[0]}
     region=${participantArray[1]}
     participantDid="did:web:did-server:$participantName"
 
-    for subject in '"region": "'$region'"' '"gaiaXMember": "true"'
-    do
-         echo "Seeding VC for $participantName: $subject"
-         vcId=$(uuidgen)
-         java -jar identity-hub-cli.jar \
-                -s="http://$participantName:8181/api/identity-hub" \
-                 vc add \
-                -c='{"id":"'$vcId'","credentialSubject":{'"$subject"}'}' \
-                -b="$participantDid" \
-                -i="did:web:did-server:gaia-x" \
-                -k="/resources/vault/gaia-x/private-key.pem"
-    done
-
-    echo "Registering $participantName"
-    java -jar registration-service-cli.jar \
-                -d="did:web:did-server:registration-service" \
-                --http-scheme \
-                -k=/resources/vault/$participantName/private-key.pem \
-                -c="$participantDid" \
-                 participants add
+  # seed  vc for participant
+  seedVerifiedCredentials "$participantName" "$participantDid" "$region"
+
+  # Register dataspace participants
+  registerParticipant "$participantName" "$participantDid"
+
+  # Wait for participant registration.
+  waitForOnboarding "$participantName" "$participantDid"
 done
+
+# flag for healthcheck by Docker
+echo "finished" > finished.flag
+echo "Finished successfully! Keep the container running."
```

Comment:
according to the docker-compose doc `--wait                      Wait for services to be running|healthy. Implies detached mode.`. This fix seems to be assuming the command waits for services to be healthy?